docs/architecture/0011-status-endpoints.md

```diff
@@ -0,0 +1,168 @@
+# 0011. Status endpoints and healthchecks
+
+Date: 2022-11-15
+
+## Status
+
+* Proposed (2022-11-15)
+
+## Context
+
+We have various endpoints for monitoring the health of the main site and its dependencies. Some of these don't appear to be used at all, while some are used (and useful). This section attempts to gather together everything we know about these endpoints, how they are currently used, and where they have gaps in their coverage.
+
+### Current endpoints
+
+The Make an LPA service has these public endpoints which describe the state of the service:
+
+* Make frontend /ping - https://lastingpowerofattorney.gov.uk/ping - returns HTML showing the status of dependencies; at the time of writing: Redis (for session data), our own Make API (see below), and DynamoDB (for the system message); **if any dependency is not OK, the status for the whole Make service is marked as not OK**. Note that the data shown for the Make API comes from the /ping endpoint on that API.
+  Used for manual checks.
+* Make frontend /ping/json - https://lastingpowerofattorney.gov.uk/ping/json - returns a JSON equivalent of the /ping endpoint.
+  Used for manual checks.
+* Make front end /ping/pingdom - https://lastingpowerofattorney.gov.uk/ping/pingdom - returns an XML equivalent (but simplified) of the /ping endpoint, in [the pingdom custom HTTP check XML format](https://www.pingdom.com/blog/new-pingdom-feature-custom-monitoring-type/).
+  Used by pingdom.
+* Make frontend nginx /robots.txt - http://`<front>`/robots.txt - not a status page as such, but used as one.
+  Used by AWS ELBs.
+
+We also have the following status pages which are only accessible internally:
+
+* Make API /ping - http://`api`:7001/ping - returns JSON showing the status of dependencies; at the time of writing: database, Sirius gateway (for tracking LPA statuses), and SQS queue (for PDF generation jobs); **if any dependency is not OK, the status for the whole API is marked as not OK**.
+  Used by the Make frontend /ping; the results are merged with the other status checks made by the front end.
+* Make API /ping/elb - http://`api`:7001/ping/elb - returns a 200 code and a short piece of text.
```

Comment:
These endpoints are accessible externally too:

https://www.lastingpowerofattorney.service.gov.uk/ping/elb

Reply:
> These endpoints are accessible externally too:
> 
> https://www.lastingpowerofattorney.service.gov.uk/ping/elb

This is another one I didn't know about :)

Reply:
@ntse is this endpoint used by the load balancer? I'm assuming not.